### ISSUE TITLE
test and validate gradient accumulation logic (+ fix if necessary)

### ISSUE THREAD

**Issue by jaidhyani** [delphi-suite/delphi#79]
Check that gradient accumulation logic works as expected. The code as is looks kind of suspect at the moment.

**Comment by jettjaniak** [comment]
thinking about a little bit more, I'm 85% confident it doesn't work
you're re-assigning a new expression to the `loss` variable in the for loop, so the previous data including gradient is lost

**Comment by jaidhyani** [comment]
But the loss variable never stored the gradient, right? The gradient is
stored on the model tensors as a side effect of `loss.backward`.

We're definitely going to add tests to validate, but I expect the current
implementation to be correct. See
https://stackoverflow.com/posts/62076913/revisions

**Comment by jettjaniak** [comment]
ha, good point!

**Comment by jaidhyani** [comment]
Factored out the gradient accumulation logic added tests for it in #89; confirmed that gradients do be accumulating.
